flask_security/decorators.py

```diff
@@ -53,7 +53,11 @@ def _get_unauthorized_view():
             except BuildError:
                 view = None
         utils.do_flash(*utils.get_message('UNAUTHORIZED'))
-        return redirect(view or request.referrer or '/')
+        redirect_to = '/'
+        if request.referrer and not request.referrer.endswith(request.path):
```

Comment:
a. disagree, since a referer to the same URL with different query arguments will not match, but still trigger a redirect loop, which is what I want to fix;
b. it's already doing that by comparing with `.endswith()`, it will work with and without hostname; in fact, the included unit test already tests both